### PR TITLE
File redirection fix

### PR DESCRIPTION
**Pull request by cogeid**
Solution to https://github.com/QData/TextAttack/issues/388

This PR solves the issue of textattack producing illegible attack logs when the output is redirected to a txt file.

Example runs from command line that demonstrates this bug fix:

`textattack attack --recipe textfooler --model bert-base-uncased-mr --num-examples 1`

This line outputs the attack logs in terminal, preserves the color method

`textattack attack --recipe textfooler --model bert-base-uncased-mr --num-examples 1 > testfile.txt`

This line outputs the attack logs in a file called testfile.txt, removes color so that it is readable

`textattack attack --recipe textfooler --model bert-base-uncased-mr --num-examples 1 --log-to-txt outputs/attacks/`

This line outputs the attack logs in a file created in the directory outputs/attacks, removes color so that it is readable

Previously the redirected output looked like this
Screen Shot 2021-06-10 at 5 51 03 PM

With this PR the redirected output would look like this, where perturbed words are highlighted by brackets
Screen Shot 2021-06-10 at 5 51 32 PM

Consequently all of the test files have also been updated with this PR.

### CODE REVIEW

**Comment by qiyanjun** [comment]
@cogeid please rebase. then I will merge.

**Comment by cogeid** [comment]
> @cogeid please rebase. then I will merge.

I have rebased

**Comment by sanchit97** [comment]
> > @cogeid please rebase. then I will merge.
> 
> I have just rebased!

I will review it.

File: textattack/loggers/attack_log_manager.py [hunk 1]
```diff
@@ -20,14 +20,17 @@ def __init__(self):
...
+    def enable_wandb(self):
+        self.loggers.append(WeightsAndBiasesLogger())
 
-    def add_output_file(self, filename):
-        self.loggers.append(FileLogger(filename=filename))
+    def disable_color(self):
+        self.loggers.append(FileLogger(stdout=True, color_method="file"))
+
+    def add_output_file(self, filename, color_method):
+        self.loggers.append(FileLogger(filename=filename, color_method=color_method))
```
Comment (sanchit97) [inline textattack/loggers/attack_log_manager.py:33]: Will the `color_method` ever not be `"file"` here?

File: textattack/attack_args.py [hunk 2]
```diff
@@ -371,7 +372,9 @@ def create_loggers_from_args(cls, args):
             if not os.path.exists(dir_path):
                 os.makedirs(os.path.dirname(txt_file_path))
 
-            attack_log_manager.add_output_file(txt_file_path)
+            # attack_log_manager.add_output_file(txt_file_path)
```
Comment (sanchit97) [inline textattack/attack_args.py:375]: I think you can remove this commented out code

**Comment by cogeid** [comment]
Yes, when the user decides not to redirect stdout to a file the color
method will be set to ansi.

File: textattack/attack_args.py (hunk 2 above)
Comment (cogeid) [reply textattack/attack_args.py]: This has been fixed

File: textattack/loggers/attack_log_manager.py (hunk 1 above)
Comment (cogeid) [reply textattack/loggers/attack_log_manager.py:33]: Yes, when the user decides not to redirect stdout to a file the color
method will be set to ansi.

Comment (sanchit97) [reply textattack/loggers/attack_log_manager.py:33]: No, I get that. However, in this `add_output_file` function, the `color_method` will always be `"file"` I think. 
Can you just change to `color_method = "file"` in L33?

**Review by sanchit97 — APPROVED** [review]
Looks good to merge

File: tests/sample_outputs/run_attack_targetedclassification2_wordnet_langtool_log-to-csv_beamsearch2_attack_n.txt [hunk 3]
```diff
@@ -55,5 +55,5 @@
 | Attack success rate:          | 100.0% |
 | Average perturbed word %:     | 2.78%  |
 | Average num. words per input: | 28.67  |
-| Avg num queries:              | 178.0  |
+| Avg num queries:              | /.*/  |
```
[the remark this answers is not in the document]
Comment (sanchit97) [reply tests/sample_outputs/run_attack_targetedclassification2_wordnet_langtool_log-to-csv_beamsearch2_attack_n.txt:58]: Does # of queries not make sense in this case?

[the remark this answers is not in the document]
Comment (cogeid) [reply tests/sample_outputs/run_attack_targetedclassification2_wordnet_langtool_log-to-csv_beamsearch2_attack_n.txt:58]: Yes this is right, it's just that I found out that the avg num queries will frequently change causing pytest to fail. I think this is due to database changing? This should prevent future pytest failing on this specific issue, but I guess the more technical approach would be to update the number each time.

**Pull request merged**